ssl/extensions.cc

```diff
@@ -3089,6 +3089,48 @@ bool ssl_negotiate_alps(SSL_HANDSHAKE *hs, uint8_t *out_alert,
   return true;
 }
 
+// Adds pha_ext to the client hello
+static bool ext_pha_add_clienthello(const SSL_HANDSHAKE *hs, CBB *out,
+                                    CBB *out_compressible, ssl_client_hello_type_t type)  {
+  const SSL *const ssl = hs->ssl;
+
+  // post_handshake_auth is not supported in TLS versions < 1.3. Also checking
+  // if extension has been enabled by client.
+  if(hs->min_version < TLS1_3_VERSION || ssl->s3->pha_ext != SSL_PHA_ENABLED) {
+    return true;
+  }
+
+  CBB contents;
+  if (!CBB_add_u16(out_compressible, TLSEXT_TYPE_post_handshake_auth) ||
+      !CBB_add_u16_length_prefixed(out_compressible, &contents) ||
+      !CBB_add_u16(&contents, 0 /* empty "extension_data" field */) ||
+      !CBB_flush(out_compressible)) {
+    return false;
```

Comment:
Should we throw an error on the stack here? I noticed OpenSSL does this.
https://github.com/openssl/openssl/blob/master/ssl/statem/extensions_clnt.c#L1203

Reply:
If adding an extension returns false, we also throw an error. This just happens in the calling context instead of the function itself: https://github.com/smittal3/aws-lc/blob/4b5c619d7a326ecd673ae6afac2774ad8fc0bb97/ssl/extensions.cc#L3470